pkg/charts/linkerd2/add-on.go

```diff
@@ -0,0 +1,29 @@
+package linkerd2
+
+import "k8s.io/helm/pkg/chartutil"
+
+var (
+	// AddOnChartsPath is where the linkerd2 add-ons will be present
+	AddOnChartsPath = "add-ons/"
+)
+
+// AddOn interface consists of the common functions required by add-ons to be implemented
+type AddOn interface {
```

Comment:
Per previous slack convo, I am hesitant about this approach where new addons are required to implement this interface. Can these values be added to the `tracing` subtree in `values.yaml`, which the CLI is already parsing? IIUC, these config are just static chart name and filepaths.

Reply:
Yep, We are getting the same values from `Values.yaml` but to parse them from `values.yaml` and create fields would need reflection? and I'm not sure if its a error-free way to do this. Having strict types enforced would not only help us with overwriting, and linkerd-values but also makes it error-free (which can be a problem with parsing.)

Reply:
I'm hoping it can be simply:
```
tracing:
  enabled: true
  chart:
    name: tracing
    templates:
       configStage: ["templates/tracing-rbac.yaml"]
       controlPlaneStage: ["templates/tracing.yaml"]
   ...
```
Then update your [`Tracing`](https://github.com/linkerd/linkerd2/blob/d8bb871d4df0af08792fec375b35523be9d0b32b/pkg/charts/linkerd2/tracing.go#L9-L14) struct to contain these fields. And in `install.go`, they should be available in your [`values`](https://github.com/linkerd/linkerd2/blob/d8bb871d4df0af08792fec375b35523be9d0b32b/cli/cmd/install.go#L804-L807) struct.

If it works, it feels cleaner than hardcoding these values in the Go code.

Reply:
Per our slack convo, we should be able to remove the `AddOn` interface by re-using some of the existing functions in the Helm library.

E.g., I am able to use `chartutil.Load()` to load the entire chart and all its dependencies metadata:
```diff
diff --git a/cli/cmd/install.go b/cli/cmd/install.go
index 752ea734..24807144 100644
--- a/cli/cmd/install.go
+++ b/cli/cmd/install.go
@@ -783,6 +783,17 @@ func render(w io.Writer, values *l5dcharts.Values) error {
 		}
 	}
 
+	mychart, err := chartutil.Load("charts/" + helmDefaultChartDir)
+	if err != nil {
+		return err
+	}
+
+	for _, d := range mychart.Dependencies {
+		for _, t := range d.Templates {
+			fmt.Printf("%s/%s\n", d.Metadata.Name, t.Name)
+		}
+	}
+
 	addons := checkAddons(values)
 
 	chart := &charts.Chart{
```

The output looks like:
```
partials/templates/NOTES.txt
partials/templates/_affinity.tpl
partials/templates/_capabilities.tpl
partials/templates/_debug.tpl
partials/templates/_helpers.tpl
partials/templates/_metadata.tpl
partials/templates/_nodeselector.tpl
partials/templates/_proxy-init.tpl
partials/templates/_proxy.tpl
partials/templates/_resources.tpl
partials/templates/_trace.tpl
partials/templates/_validate.tpl
partials/templates/_volumes.tpl
tracing/templates/tracing-rbac.yaml  # tracing's chart name + template file
tracing/templates/tracing.yaml  # tracing's chart name + template file
```

Does this give us what we want?

Reply:
Yep, Updated now